src/dataflow/components/dataflow-program-topbar.tsx

```diff
@@ -12,43 +12,67 @@ interface TopbarProps {
   onProgramTimeSelectClick: (type: number) => void;
   isRunEnabled: boolean;
   runningProgram: boolean;
+  remainingTime: number;
   readOnly: boolean;
 }
 
 export const DataflowProgramTopbar = (props: TopbarProps) => {
   const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
     props.onProgramTimeSelectClick(Number(event.target.value));
   };
+  const runTime = props.programRunTimes.find( (rt: ProgramRunTime) => rt.val === props.programDefaultRunTime );
+  const remainingHours = Math.floor(props.remainingTime / 3600);
+  const remainingMinutes = Math.floor((props.remainingTime - remainingHours * 3600) / 60);
+  const remainingSeconds = (props.remainingTime - (remainingHours * 3600) - (remainingMinutes * 60) ) % 60;
+  const progressWidth = (76 * props.remainingTime / props.programDefaultRunTime).toString() + "px";
+  const hoursString = String(remainingHours).padStart(2, "0");
+  const minutesString = String(remainingMinutes).padStart(2, "0");
+  const secondsString = String(remainingSeconds).padStart(2, "0");
   return (
     <div className="program-editor-topbar">
-      <div className="duration" title="Duration">
-        <div className="label-back">
-          <div className="label">Duration</div>
-        </div>
-        <div className="duration-options-back">
-          <HTMLSelect className="duration-options"
-            onChange={handleSelectChange}
-            disabled={!props.isRunEnabled || props.readOnly}
-            value={props.programDefaultRunTime.toString()}
-          >
-            { props.programRunTimes.map((rt: ProgramRunTime) => (
-                <option key={rt.text} value={rt.val}>
-                  {rt.text}
-                </option>
-              ))
-            }
-          </HTMLSelect>
-        </div>
-      </div>
-      <button
-        className="program-state-button"
-        title="Run"
-        onClick={props.onRunProgramClick}
-        disabled={!props.isRunEnabled || props.readOnly}
-      >
-        <div className="icon run" />
-        <div className="text">Run</div>
-      </button>
+      { props.runningProgram
```

Comment:
For clarity, I'd consider separating this into a local `CountdownTimerComponent` and a `DurationSelectorComponent` (or names to that effect).

Reply:
Split these apart into 2 local components.